spec/filters/dns_spec.rb

```diff
@@ -4,357 +4,391 @@
 require "resolv"
 
 describe LogStash::Filters::DNS do
-  before(:each) do
-    allow_any_instance_of(Resolv).to receive(:getaddress).with("carrera.databits.net").and_return("199.192.228.250")
-    allow_any_instance_of(Resolv).to receive(:getaddress).with("does.not.exist").and_raise(Resolv::ResolvError)
-    allow_any_instance_of(Resolv).to receive(:getaddress).with("nonexistanthostname###.net").and_raise(Resolv::ResolvError)
-    allow_any_instance_of(Resolv).to receive(:getname).with("199.192.228.250").and_return("carrera.databits.net")
-    allow_any_instance_of(Resolv).to receive(:getname).with("127.0.0.1").and_return("localhost")
-    allow_any_instance_of(Resolv).to receive(:getname).with("128.0.0.1").and_raise(Resolv::ResolvError)
-    allow_any_instance_of(Resolv).to receive(:getname).with("199.192.228.250").and_return("carrera.databits.net")
-  end
+  describe "with stubbed Resolv" do
+    before(:each) do
+      allow_any_instance_of(Resolv).to receive(:getaddress).with("carrera.databits.net").and_return("199.192.228.250")
```

Comment:
I was planning on getting this PR merged then create an issue to use a fixtures hosts file to replace these stubs - because I think we could be missing bugs that are triggered from data returning from the Stdlib Resolve class.